zerver/views/custom_profile_fields.py

```diff
@@ -229,24 +274,17 @@ def update_realm_custom_profile_field(
 
     if (
         field.field_type == CustomProfileField.EXTERNAL_ACCOUNT
-        # HACK: Allow changing the display_in_profile_summary property
-        # of default external account types, but not any others.
-        #
-        # TODO: Make the name/hint/field_data parameters optional, and
-        # just require that None was passed for all of them for this case.
```

Comment:
I have not changed the conditions or the error being raised here. I think the error should be raised when someone tries to update the name, hint, field_data of a default custom field. Still not entirely sure what the TODO intended, so removed it from what info I have, please LMK if I can make any code changes to address the TODO

Reply:
Got it now, made the comment a bit more clearer in that case: https://github.com/zulip/zulip/compare/23e1a46637cecc94cc71adc31bf3f0e544871be8..4bc5e8d101762406f832e12ee2eb298042f407de